Update to latest Lua 5.4 version

dsmgit opened the pull request:
Here are the changes I made to the std library. It's complete now. Though, I'd like to ask if u can verify the params and return types for rightness. We don't want to miss something. Also, the built-in types are added as classes, they're objects. Why are they written in small caps? I'm working on a API and those have types written in pascal case (like Boolean, Number, String etc.), because they're objects. Thank you in advance!

tangzx commented on src/main/resources/std/string.lua:
```diff
@@ -1,114 +1,138 @@
+--- a way adequate to your needs.)
 ---@param func fun()
-function string.dump(func) end
+---@param optional strip boolean
```
`optional` is deprecated. use `@overload` instead of.

```
---@overload fun(func:fun()):string
---@param func fun()
---@param strip boolean
---@return string
function string.dump(func, strip) end
```

dsmgit replied:
Ow, okay! I'll change the appearance of optional in the code. What about the types. Do they need to stay in small caps?

tangzx replied:
Yes,  the built-in (primitive) types written in lower case, just like c#/as3/...

dsmgit replied:
Okay, C# also supports OOP types - in CamelCase, but I'll leave it all in lower.

tangzx commented:
Parameters and return type verification was a existing feature, disabled by default, this is experimental with many issues. enable it via: settings-> editor -> inspections -> Lua>Lint>Type safety

dsmgit commented:
I could if the latest EAP supports the plugin. Haven't done anything with Lua for two days now.

tangzx commented:
The latest daily build support 2018.2-EAP.
https://ci.appveyor.com/project/EmmyLua/intellij-emmylua

dsmgit commented:
I tried to install the plugin on 2018.2-EAP 182.2371.4 build, but it doesn't show up in the plugins dialog.

dsmgit commented:
The optional params are fixed now. You can check the params, types etc. for rightness, before merging. 

I moved the _G and the _VERSION globals into global.lua. Also added the full type list supported as of 5.4, where replaced the "void" type with "nil". And,, in debug.lua where it says "local" I replace that as "var", since "local" is a reserved keyword and can't be used as parameter.

IntendedConsequence commented:
Does it mean that EmmyLua is gonna support code completion only for Lua 5.4 stdlib now?
